src/sys/mman.rs

```diff
@@ -324,6 +324,21 @@ pub fn munlockall() -> Result<()> {
     unsafe { Errno::result(libc::munlockall()) }.map(drop)
 }
 
+#[cfg(any(target_os = "linux"))]
+pub type mincore_vec_char_t = std::os::raw::c_uchar;
+#[cfg(any(target_os = "freebsd", target_os = "macos"))]
+pub type mincore_vec_char_t = std::os::raw::c_char;
+
+/// Inspect page residency in memory at the address `addr` continuing for `length` bytes.
+///
+/// # Safety
```

Comment:
`Safety` sections normally only apply to `unsafe` functions.  Should this function be `unsafe`?  What would happen if the user supplies invalid arguments?

Reply:
In new suggested signature, I suppose there's no unsafety in this method. I'll fix doc comment.
https://github.com/nix-rust/nix/pull/1530#discussion_r711755282